libweston/backend-rdp/rdprail.c

```diff
@@ -3191,6 +3191,44 @@ disp_client_monitor_layout_change(DispServerContext *context, const DISPLAY_CONT
 	return CHANNEL_RC_OK;
 }
 
+static void
+rdp_rail_idle_handler(struct wl_listener *listener, void *data)

```

Comment:
Could this and the wake handler be combined into one handler for wake and idle? It looks to me like the void* data passed is the compositor and the compositor state is updated to WESTON_COMPOSITOR_ACTIVE/WESTON_COMPOSITOR_IDLE before the signal is emitted which could be checked in the function. Or is there a reason to separate them? Just wondering because I'm curious.

Reply:
Yes, you are right, it can be consolidated into single function, the reason having separate callback is that's how it's done in other components from upstream. I think I would like to maintain same pattern as upstream.

Reply:
I see, that makes sense! Thanks for explaining. Looks good to me!